File: mybatis-plus-sample-deluxe/src/main/java/com/baomidou/mybatisplus/samples/deluxe/MyMetaObjectHandler.java
package com.baomidou.mybatisplus.samples.deluxe;

import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler;
import org.apache.ibatis.reflection.MetaObject;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.sql.Timestamp;

/**
 * 填充器
 *
 * @author nieqiurong 2018-08-10 22:59:23.
 */
@Component
public class MyMetaObjectHandler implements MetaObjectHandler {

    private static final Logger LOGGER = LoggerFactory.getLogger(MyMetaObjectHandler.class);

    @Override
    public void insertFill(MetaObject metaObject) {
        LOGGER.info("start insert fill ....");
        //避免使用metaObject.setValue()
        this.strictInsertFill(metaObject, "createTime", Timestamp.class, new Timestamp(System.currentTimeMillis()));
    }

    @Override
    public void updateFill(MetaObject metaObject) {
        LOGGER.info("nothing to fill ....");
    }
}

File: mybatis-plus-sample-kotlin/src/main/kotlin/com/baomidou/mybatisplus/samples/kotlin/config/MyMetaObjectHandler.kt
package com.baomidou.mybatisplus.samples.kotlin.config

import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler
import org.apache.ibatis.reflection.MetaObject
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import java.time.LocalDateTime

/**
 * @author nieqiurong
 */
@Component
class MyMetaObjectHandler : MetaObjectHandler {

    private val log = LoggerFactory.getLogger(MyMetaObjectHandler::class.java)

    // 注意将kotlin类型转为java类型请使用 xxx::class.javaObjectType，防止部分类型使用xxx::class.java转换为基本类型导致类型不一致无法填充
    override fun insertFill(metaObject: MetaObject) {
        log.info("开始插入填充...");
        this.strictInsertFill(metaObject, "createUserId", Long::class.javaObjectType, 1111L)
        this.strictInsertFill(metaObject, "createTime", LocalDateTime::class.javaObjectType, LocalDateTime.now())
    }

    override fun updateFill(metaObject: MetaObject) {
        log.info("开始更新填充...");
        this.strictUpdateFill(metaObject, "updateUserId", Long::class.javaObjectType, 6666L)
        this.strictUpdateFill(metaObject, "updateTime", LocalDateTime::class.javaObjectType, LocalDateTime.now())
    }

}